src/backend/access/transam/xlog.c

```diff
@@ -6169,12 +6169,16 @@ SetLastWrittenLSNForBlockRange(XLogRecPtr lsn, RelFileNode rnode, ForkNumber for
 		BufferTag key;
 		bool found;
 		BlockNumber bucket;
+		BlockNumber start_bucket; /* inclusive */
+		BlockNumber end_bucket;   /* exclusive */
+
+		start_bucket = from / LAST_WRITTEN_LSN_CACHE_BUCKET;
+		end_bucket = n_blocks == 0
```

Comment:
Could do the `n_blocks == 0` earlier in the function, like we check for `lsn == InvalidXLogRecPtr`. Might be more clear.